### PR TITLE
Not generate deps and config file for native aot apps

### LINKED ISSUE #63265: Publishing a NativeAOT app generates JSON files
### Description

Publishing a console app as AOT through VS still generates the following JSON files:

```
AppName.runtimeconfig.json
AppName.deps.json
```

If you look at these files it tells something about the .NET version. The .NET references etc. All of this seems pretty much obsolete for a native executable. Is there any reason why these files are generated or is this a bug?

### Reproduction Steps

1. Create console .NET 6 app
2. Add ILCompiler nuget package
3. In VS 2022 create a publish to folder profile with the following settings: Self-contained, win-x64
4. Publish through VS

### Expected behavior

Only executable in publish folder

### Actual behavior

Executables + JSON files in publish folder

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by LakshanF**
Fixes #63265

### CODE REVIEW

File: src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.targets [hunk 1]
```diff
@@ -37,6 +37,10 @@
     <IlcSetupPropertiesDependsOn>ImportRuntimeIlcPackageTarget</IlcSetupPropertiesDependsOn>
     <IlcDynamicBuildPropertyDependencies>SetupProperties</IlcDynamicBuildPropertyDependencies>
 
+    <!-- NativeAOT apps do not require deps.json and runtimeconfig.json files -->
+    <GenerateDependencyFile>false</GenerateDependencyFile>
+    <GenerateRuntimeConfigurationFiles>false</GenerateRuntimeConfigurationFiles>
```
Comment (vitek-karas) [inline src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.targets:42]: Test?

**Review by vitek-karas — COMMENTED** [review]
This reminded me - SDK supports customizing `.runtimeconfig.json` via a `runtimeconfig.template.json` - this mechanism won't work for NativeAOT. Would be good to warn/fail if the template is specified since we're going to "ignore" it.
This is already the case regardless of this change - maybe just file an issue...

File: src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.targets (hunk 1 above)
Comment (LakshanF) [reply src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.targets:42]: I created an issue for adding a test in the sdk repo for this, https://github.com/dotnet/sdk/issues/26132

Comment (elinor-fung) [inline src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.targets:42]: I thought NativeAOT was a publish step. Won't this also affect build and make the build output incomplete?

Comment (LakshanF) [reply src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.targets]: Moved the properties under a target that will be invoked only during publish.

**Review by sbomer — APPROVED** [review]
LGTM but I would make sure @MichalStrehovsky gets a chance to review it.

**Comment by MichalStrehovsky** [comment]
Having this in a target that changes the value of the property mid-publish is a bit of a hack. We used hacks like this before Native AOT became a shipping product. The SDK is aware of various publishing details. I think it will be better to do this in the SDK.

The SDK decides that we should copy these files in `ComputeResolvedFilesToPublishList` - I think we can put those two items in a separate `ItemGroup` that is conditioned on not being `PublishAot`.

It sets us up towards undoing other hacks we have in NativeAOT targets (we have a target that runs after `Publish` that deletes stuff from the publish and copies the native bits - those are other things that can now be patched up in `ComputeResolvedFilesToPublishList` directly). But that part is for a different pull request.

**Pull request closed without merging**